src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/TypeNameHelper.cs

```diff
@@ -1,83 +1,96 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#nullable disable
-
 using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Collections.Immutable;
 using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
 
 namespace Microsoft.AspNetCore.Razor.Language;
-internal class TypeNameHelper
+
+internal static class TypeNameHelper
 {
-    private static readonly string[] PredefinedTypeNames = new[] {
+    private const string GlobalPrefix = "global::";
+
+    private static readonly ImmutableArray<string> PredefinedTypeNames = new[]
```

Comment:
I wonder why this is not a set since we are just checking for `Contains`.

Reply:
Originally, I was pushing everything to `ReadOnlySpan<char>` and was running into problems with it being a ref struct. Now that I've normalized to `ReadOnlyMemory<char>`, I think this can be switched. Thanks!

Reply:
Actually, I'm not so sure. The hash code for `ReadOnlyMemory<char>` includes the index and length. So, this might be a bit of a quagmire.

Reply:
Also, `ReadOnlySpan<char>` compares the internal string with reference equality, not value equality. I guess a special EqualityComparer is needed.